tilavarauspalvelu/admin/email_template/forms.py

```diff
@@ -162,384 +149,432 @@ class PendingInstructionsFormMixin(forms.Form):
     pending_instructions = forms.CharField(initial="[KÄSITELTÄVÄN VARAUKSEN OHJEET]", widget=text_area_widget)
 
 
-# --- Tester forms --------------------------------------------------------------------------------------------------
+# --- Base forms -------------------------------------------------------------------------------------------------
 
 
-class ApplicationHandledEmailTemplateTesterForm(
-    LanguageFormMixin,
-    BaseEmailTemplateForm,
-):
+class BaseEmailTemplateForm(LanguageFormMixin, forms.Form):
+    """Base form for email template testers, not meant to be used directly."""
+
+    send_to = forms.EmailField(initial="", widget=email_widget)
+
+    @classmethod
+    def from_reservation_unit(cls, instance: ReservationUnit, *, language: Lang) -> Self:
+        """Initialise the form with data form from model information."""
+        return cls(initial=cls.get_initial_data_from_reservation_unit(instance, language=language))
+
+    @classmethod
+    def get_initial_data_from_reservation_unit(cls, instance: ReservationUnit, *, language: Lang) -> dict[str, Any]:  # noqa: ARG003
+        """Get initial data from model information."""
+        return {}
+
     def to_context(self) -> EmailContext:
-        return get_context_for_application_handled(language=self.cleaned_data["language"])
+        """Convert form data to email context."""
+        raise NotImplementedError
+
+    def get_context_params(self) -> dict[str, Any]:
+        return {
+            "language": self.cleaned_data["language"],
+        }
 
 
-class ApplicationInAllocationEmailTemplateTesterForm(
-    LanguageFormMixin,
+class ReservationBaseForm(
+    ReservationBasicInfoFormMixin,
+    EmailRecipientFormMixin,
     BaseEmailTemplateForm,
 ):
+    @classmethod
+    def get_initial_data_from_reservation_unit(cls, instance: ReservationUnit, *, language: Lang) -> Self:
```

Comment:
Return typing is wrong for this method in many of the subclasses. Should be `dict[str, Any]` like the superclass method.